web/controllers/pair_controller.ex

```diff
@@ -18,25 +18,31 @@ defmodule Pairmotron.PairController do
     {year, _} = y |> Integer.parse
     {week, _} = w |> Integer.parse
     user = conn.assigns[:current_user]
-    groups_and_pairs = fetch_pairs(year, week, user)
+    groups_and_pairs = fetch_pairs(conn, year, week, user)
     conn = assign_current_user_pair_retro_for_week(conn, year, week)
     render conn, "index.html", year: year, week: week, groups_and_pairs: groups_and_pairs,
       start_date: Timex.from_iso_triplet({year, week, 1}),
       stop_date: Timex.from_iso_triplet({year, week, 7})
   end
 
-  defp fetch_pairs(year, week, user) do
+  defp fetch_pairs(conn, year, week, user) do
     user = user
       |> Repo.preload(:groups)
     pairs = user
-      |> pairs_for_user_groups(year, week)
+      |> pairs_for_user_groups(conn, year, week)
     Enum.zip(user.groups, pairs)
   end
 
-  defp pairs_for_user_groups(user, year, week) do
+  defp pairs_for_user_groups(user, conn, year, week) do
     user.groups
       |> Enum.map(fn(group) ->
-        PairMaker.fetch_or_gen(year, week, group.id)
+        case PairMaker.fetch_or_gen(year, week, group.id) do
+          {:error, pairs, message} ->
+            conn |> put_flash(:error, message)
```

Comment:
Same here. All put_flash is doing is adding to a property on the conn, which isn't actually returning to anywhere (since this case is returning pairs).